library/src/scala/tasty/reflect/TreeUtils.scala

```diff
@@ -98,6 +98,7 @@ trait TreeUtils
       case TypeTree.ByName(result) => foldTypeTree(x, result)
       case TypeTree.Annotated(arg, annot) => foldTree(foldTypeTree(x, arg), annot)
       case TypeBoundsTree(lo, hi) => foldTypeTree(foldTypeTree(x, lo), hi)
+      case SyntheticBounds() => x
```

Comment:
What is `SyntheticBounds`?

Reply:
Internally it is a `TypeTree` that has a type bound. We needed two different extractors `Synthetic`/`SyntheticBounds` as we differentiate separate abstract types for types and type bounds.

Reply:
You mean a "type" of the for `_ >: L <: U`? If yes, should it be called `WildcardType` then? `SyntheticBounds` tells me not a lot.

Reply:
Yes, it is for wildcards but aslo for some synthetic bounds.

https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/tastyreflect/TypeOrBoundsTreesOpsImpl.scala#L162-L168

After some exploration I discovered that the second case only apears with enums. It looks like we generate some type parameter bounds as a `TypeTree` wheras all other cases are in a `TypeBoundsTree`.

I will rename it to `WildcardType` and move the other case to `TypeBoundsTree`.